src/elements/forms/text-field.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import textFieldClassNames from './text-field.css'
+
+import { classNames } from 'utils'
+
+const generateId = () =>
+  // Math.random should be unique because of its seeding algorithm.
+  // Convert it to base 36 (numbers + letters), and grab the first 9 characters
+  // after the decimal.
+  Math.random()
+    .toString(36)
+    .substr(2, 9)
+
+const TextField = ({
+  children,
+  id,
+  className,
+  label,
+  size,
+  tag: Tag = 'div',
+  ...inputProps
+}) => {
+  const controlId = [id, 'control'].join('-')
+
+  return (
+    <Tag
+      id={id}
+      className={classNames
+        // prevents passing 'default' to the class list
+        .use('container', { [size]: size !== 'default' }, className)
+        .withModule(textFieldClassNames)}
+    >
+      <input id={controlId} {...inputProps} />
+      <label htmlFor={controlId}>{label}</label>
```

Comment:
It would be nice to render this element only when the label prop is provided.

Reply:
No, not really. It must always exist. It may be for screen-readers only but I do not know how to pass a parameter for it. Any suggestions?

Reply:
Do you mean like parameter for screen-readers only?

Reply:
Yes. This is the only need.

Also, if you want an input element without label, you can just use `<input>` instead.

Reply:
I would simply copy-paste .sr-only class from bootstrap and add a new prop to TextField component like `labelsronly=false` which would conditionally add this class to label element.